pkg/model/health.go

```diff
@@ -0,0 +1,94 @@
+package model
+
+import (
+	"log"
+	"strconv"
+
+	"github.com/Merteg/pl-health-service/proto"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+)
+
+type Health struct {
+	Id         string             `json:"_id,omitempty", bson: "_id, omitempty"`
+	TargetID   string             `json:"targetid,omitempty" validate:"required", bson:"targetid, omitempty" validate:"required"`
+	TargetType string             `json:"targettype,omitempty" validate:"required", bson:"targettype, omitempty" validate:"required"`
+	Status     string             `json:"healthstatus,omitempty",  bson:"healthstatus, omitempty"`
+	Counters   map[string]int32   `json:"counters,omitempty",  bson:"counters, omitempty"`
+	Metrics    map[string]float64 `json:"metrics,omitempty",  bson:"metrics"`
+	Hearthbeat bool               `json:"heartbeat,omitempty",  bson:"heartbeat, omitempty"`
+	Messages   []*Message         `json:"messages,omitempty", bson:"messages, omitempty"`
+	Timestamp  int64              `json:"timestamp,omitempty" validate:"required", bson:"timestamp, omitempty" validate:"required"`
+}
+type Message struct {
+	Summary        string             `json:"Summary"`
+	Error          string             `json:"Error"`
+	AffectedHealth bool               `json:"AffectedHealth"`
+	Status         proto.HealthStatus `json:"Status"`
+}
+
+func (h *Health) FromProto(health *proto.Health) {
+	heartbeat := health.GetHearthbeat()
+	msgs := make([]*Message, 0, len(health.Messages))
+	for _, message := range health.Messages {
+		msgs = append(msgs, MessageFromProto(message))
+	}
+
+	h.TargetID = health.GetTargetID()
+	h.TargetType = health.GetTargetType()
+	h.Status = health.GetStatus().String()
+	h.Hearthbeat = heartbeat.Value

```

Comment:
check if heartbeat is not null. You will have null pointer here in case it is null